doc/sphinx/proof-engine/categories.txt

```diff
@@ -0,0 +1,58 @@
+A first stab at categories:
+
+Values
+  Types of values
+  Syntactic values
+  Tactics in terms
+  Substitution
+  Local definitions: let
+  Function construction and application
+
+Goal selectors (existing subsection)
+
+Processing multiple goals (existing subsection)
+
+    (The tactic monad, including sequencing, backtracking, control flow, success and failure,
+    and focusing on specific goals):
+Backtracking and branching (existing subsection)
+
+Control flow
+  Sequence: ;
+  Do loop
+  Repeat loop
+  Catching errors: try
+  Conditional branching: tryif
+  Testing boolean expressions: guard
+
+Alternatives
+  Branching with backtracking: +
+  Local application of tactics: [> ... ]
+  First tactic to succeed
+  Solving
+  First tactic to make progress: ||
+  Detecting progress
+
+Success and failure
+  Checking for success: assert_succeeds
+  Checking for failure: assert_fails
+  Failing
+  Soft cut: once
+  Checking for a single success: exactly_once
+
+Manipulating values
```

Comment:
Does `constr:(...)`, `open_constr:(...)`, and `uconstr:(...)` go somewhere here?

Reply:
Those are in "Syntactic values".  The earlier section introduces the various types of values.  IMHO it's common to introduce values before control structures when describing a language.  I suppose the Values section could go just before "Manipulating values" as it is less central to Ltac.  WDYT?